redaxo/src/core/lib/sql/sql.php

```diff
@@ -1908,7 +1908,7 @@ public static function checkDbConnection($host, $login, $password, $database, $c
                             $login,
                             $password
                         );
-                        if (1 !== $conn->exec('CREATE DATABASE ' . $database . ' CHARACTER SET utf8 COLLATE utf8_general_ci')) {
+                        if (1 !== $conn->exec('CREATE DATABASE ' . $database . ' CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci')) {
```

Comment:
Von `_general_ci` auf `_unicode_ci` finde ich gut.
Darum ging es ja auch nur in der zitierten Changelog-Meldung, dass wir nun `utf8_unicode_ci` und `utf8mb4_unicode_ci` verwenden.

Frage für mich ist, ob wir hier auch wirklich schon auf `utf8mb4_` gehen können/sollen.
Ich glaube aber, dass es ok ist. Wenn man mit `rex_sql_table` arbeitet (was core und core-addons tun), dann ist es sowieso egal, wie die Datenbank eingestellt ist.

Reply:
Wenn man sieht, wie gelegentlich nur durch search_it auffällt, wie viele noch MySQL 5.5 im Einsatz haben, dann könnte das viele aus der Bahn werfen, weil sie R u.U. nicht mehr dort installieren können.

utf8mb4 benötigt MySQL 5.6.

(Persönliche Meinung: alte Zöpfe abschneiden. 👍)

Reply:
MySQL 5.5 unterstützen wir schon seit ein paar Versionen nicht mehr.
Aber utf8mb4 unterstützt der Core sogar erst ab 5.7.7.
Wenn man im nächsten Setup-Schritt utf8 auswählt, dann werden trotzdem alle Tabellen als utf8 angelegt. Somit auch möglich bei MySQL 5.6.

Die Datenbank-Einstellung ist nur relevant für Addons, die Tabellen nicht über `rex_sql_table` anlegen, und ohne explizite charset-Angabe. Und wenn man Tabellen manuell selbst anlegt.

Meist kommt diese Zeile sowieso gar nicht zum Tragen, da die Datenbank schon existiert.

Reply:
Richtig, für das Gros wird dieser Punkt nicht relevant sein. Konsequenterweise sollte man jedoch (zumindest meiner Meinung nach) die Installationsvoraussetzungen auf den in der Addon-Programmierung gesetzten Standard heben. Meine Datenbank ist "general" gewesen und mir hat es alle naselang Fehler auf den Schirm gezaubert. Letztes Beispiel "FOR Aufgaben", hier wird auf Relationen mit der user-Tabelle zugriffen, die eben wegen der Kollation-Unterschiede in einen Wow-Screen laufen. 
Perse laufen alle "nativ"-angelegten Tabellen im Kollations-Kontext der Datenbank. Heben ich also alle Kern-Addons auf utf8mb4 sollte es doch hinreichende Bedingung für einen Standard sein. Damit auch die nicht rex_sql_tables - Tabellen sicher sind. Andersherum wäre es, für mein Dafürhalten, wesentlich problematischer.